apps/s_time.c

```diff
@@ -270,10 +270,10 @@ int s_time_main(int argc, char **argv)
     /* Loop and time how long it takes to make connections */
 
     bytes_read = 0;
-    finishtime = (long)time(NULL) + maxtime;
+    finishtime = (long)((unsigned long)time(NULL) + maxtime);
```

Comment:
AFAICT the local variables `finishtime` and `maxtime` are not part of the API or ABI.  Can't we change their type (and other internal variables' type) to be native `time_t` and only cast at the boundaries where we have to conform to existing API contracts?

Reply:
Yes, that makes sense, I am also tempted to reverse the computation and store a starttime instead of the finishtime, that needs also lots of type casts, but would also be robust if time is set back.

Reply:
I would advise using a separate PR for the "store starttime instead of finishtime" change.